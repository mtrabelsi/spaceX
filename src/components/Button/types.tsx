export type ButtonPropType = {
    buttonStyle?: object,
    disabled?: boolean,
    iconLeft?: any,
    textStyle?: object,
    imgStyle?: object,
    value?: string,
    hasIconLeft?: boolean,
    start?: boolean,
    simpleMode?: boolean,
    special?: boolean,
    onClick?: (e: React.MouseEvent) => void,
    isDanger?: boolean
}
