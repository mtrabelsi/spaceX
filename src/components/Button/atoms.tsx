import s from 'styled-components';
import {
  fontFamily,
  secondaryFontSize,
  backgroundImage,
  veryLightColor,
  defaultBoxShadow,
  simpleBoxShadow,
  endColor,
  errorColor,
} from '../Global';
import { ButtonPropType } from './types';

const decideBackground = (p: ButtonPropType) => {
  if (p.simpleMode) {
    return endColor;
  } if (p.isDanger) {
    return errorColor;
  }
  return backgroundImage;
};
export const Button = s.button<ButtonPropType>`
  opacity: ${(p) => (p.disabled ? 0.3 : 1)};
  cursor: ${(p) => (p.disabled ? 'none' : 'pointer')};
  pointer-events: ${(p) => (p.disabled ? 'none' : 'initial')};
  box-shadow: ${(p) => (p.simpleMode ? simpleBoxShadow : defaultBoxShadow)};
  background: ${(p) => decideBackground(p)};
  border: none;
  color: ${veryLightColor};
  padding: 5px 0;
  outline: none;
  min-width: 200px;
  height: 40px;
  border-radius: 3px;
  position: relative;
  font-family: ${fontFamily};
  text-transform: uppercase;
  font-weight: bold;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const ButtonText = s.span`
  font-size: ${secondaryFontSize};
  font-family: ${fontFamily};
  font-style: normal;
  text-transform: uppercase;
`;
