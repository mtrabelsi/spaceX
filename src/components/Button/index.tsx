import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { Button, ButtonText } from './atoms';
import { Img } from '../InputText/atoms';
import { ButtonPropType } from './types';

const Btn: React.FC<ButtonPropType | RouteComponentProps> = (props) => {
  const {
    disabled,
    buttonStyle,
    simpleMode,
    isDanger,
    iconLeft,
    hasIconLeft,
    value,
    imgStyle,
    textStyle,
    onClick,
  } = props as ButtonPropType;
  return (
    <Button
      onClick={onClick}
      simpleMode={simpleMode}
      disabled={disabled}
      isDanger={isDanger}
      style={buttonStyle}
    >
      {hasIconLeft && iconLeft && <Img src={iconLeft} style={imgStyle} />}
      <ButtonText style={textStyle}>{value}</ButtonText>
    </Button>
  );
};

export default Btn;
