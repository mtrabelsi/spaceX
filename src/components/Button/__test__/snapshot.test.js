import React from 'react';
import renderer from 'react-test-renderer';
import Button from '..';

it('Should consider all props and generate unique html based on that', () => {
  const tree = renderer
    .create(<Button
      buttonStyle={{ color: 'red', padding: 5 }}
      imgStyle={{ width: 20 }}
      value="button value"
      hasIconLeft
      iconLeft="<"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
