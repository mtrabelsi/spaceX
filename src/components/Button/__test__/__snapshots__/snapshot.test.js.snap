// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should consider all props and generate unique html based on that 1`] = `
<button
  className="sc-AxjAm fLhDvz"
  style={
    Object {
      "color": "red",
      "padding": 5,
    }
  }
>
  <img
    className="sc-AxhUy ionFUO"
    src="<"
    style={
      Object {
        "width": 20,
      }
    }
  />
  <span
    className="sc-AxirZ iRauBN"
  >
    button value
  </span>
</button>
`;
